stopwatch.py

```diff
@@ -206,7 +210,10 @@ def timer(self, name, bucket=None, start_time=None, end_time=None):
             self.add_annotation('Exception', type(e).__name__, event_time=end_time)
             raise
         finally:
-            self.end(name, end_time=end_time, bucket=bucket)
+            if name in self._cancelled_spans:
+                del self._cancelled_spans[name]
+            else:
+                self.end(name, end_time=end_time, bucket=bucket)
```

Comment:
Can we roll this check into self.end(). Feels like we should always check cancelled_spans on end(), not just within a context manager.